src/com/company/Main.java

```diff
@@ -1,89 +1,70 @@
 package com.company;
+
 import java.io.File;
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Paths;
 import java.util.StringTokenizer;
-import java.lang.String;
-
 
 public class Main {
-    public static void main(String[] args){
+    final static String FILE_TYPE_TXT = ".txt";
+    final static String FILE_TYPE_JSON = ".json";
+    final static String SPECIAL_CHAR_REMOVAL_REGEX = "[^a-zA-Z0-9@]";
+    final static String SINGLE_SPACE = " ";
 
-        String txtfile = args[0];
-        File file = new File(txtfile);
-        checkforsupport(file);
-        checkforexistance(file);
+    public static void main(String[] args) {
+
+        if (args.length != 2) {
+            System.out.println("Exiting, since no file path and word to search for are provided!");
+            return;
+        }
+        String txtFilePath = args[0];
         String searchword = args[1];
-        System.out.println("processing................");
-        String data = null;
-        data = readFileAsString(txtfile);
-        avoidspecialchar(data);
-        searchtheword(data,searchword);
+        File file = new File(txtFilePath);
 
+        if (!file.exists() || !isFileFormatSupported(file)) {
+            System.out.println("File is not valid");
+            return;
+        }
+        if (file.length() == 0) {
+            System.out.println("The file does not contains any data ");
+            return;
+        }
+        System.out.println("Processing................");
+        String data = readFileAsString(txtFilePath);
 
+        searchTheWord(data, searchword);
     }
 
-
-        public static String readFileAsString(String fileName)
-             {
-                String data = "";
-              try {
-                   data = new String(Files.readAllBytes(Paths.get(fileName)));
-              } catch (IOException e) {
-                  e.printStackTrace();
-              }
-              return data;
+    public static String readFileAsString(String fileName) {
+        String data = null;
+        try {
+            data = new String(Files.readAllBytes(Paths.get(fileName)));
+            data.replaceAll(SPECIAL_CHAR_REMOVAL_REGEX, SINGLE_SPACE);
```

Comment:
Shouldn't it be data=data.replaceAll() ? Before changing to this try to read a file that contains ',' in the text and see the output.

Reply:
yeah checked that
data replaced was not storing in the variable data